### DISCUSSION TITLE
3.5 pre-release notes

### DISCUSSION THREAD

**Discussion by kjk** [sumatrapdfreader/sumatrapdf#2735] [Announcements]
This is a list of changes in latest [pre-release builds](https://www.sumatrapdfreader.org/prerelease):
* ability to drag out a tab to open it in new window
* support opening `.avif` images (including inside .cbz/,cbr files)
* respect image orientation exif metadata in .jpeg and .png images
* support Adobe Reader syntax for opening files `/A "page=<pageno>#nameddest=<dest>search=<string>`
* add `Next Tab` / `Prev Tab` commands and bind them to `Ctrl + PageUp` / `Ctrl + PageDown`
* keep Home tab open; add `NoHomeTab` advanced option to disable that
* add context menu to tabs
* bugfix: handle files we can't open in `next file in folder` / `prev file in folder` commands
* command palette: when search starts with `>`, only show commands, not files (like in Visual Studio Code)
* add `reopen last closed` command (`Ctrl + Shift + T`, like in web browsers)
* add `clear history` command
* can send commands via [DDE](https://www.sumatrapdfreader.org/docs/DDE-Commands)
* added `CmdOpenWithExplorer`, `CmdOpenWithDirectoryOpus`, `CmdOpenWithTotalCommander`, `CmdOpenWithDoubleCommander` commands
* enable `CmdCloseOtherTabs`, `CmdCloseTabsToTheRight` commands from command palette
* recognize `PgUp` / `PgDown` in keyboard shortcuts
* add `-disable-auto-rotation` cmd-line print option
* Arm 64 builds

Also in https://www.sumatrapdfreader.org/docs/Version-history

**Comment by ilyaza** [comment]
As well as I understand, the `Ctrl + PageUp` / `Ctrl + PageDown` replaces the functionality of old `Ctrl-Tab` / `Ctrl-Shift-Tab`?

If so, then the binding of `Ctrl-Tab`  is free now — so can it be implemented to work “as a good browser”?  Please?!

E.g., in `FireFox`, these move through the tabs (in the given window) in ***the order of last chronological usage*** (and this order is not updated until `Ctrl` is released).  For example, `Ctrl-Tab-Tab-Tab`  moves to the fourth tab in this order, while `Ctrl-Tab-Tab-Tab-Shift-Tab` moves to the third one.  

(It is also important to be able to cancel by `Esc` before `Tab` is released…  For an “extra touch”, one may allow `Left` / `Right` to be aliases for `Tab` and `Shift-Tab` before `Ctrl` is released!)

**Comment by ilyaza** [comment]
This also bring another question: what is the order of tabs in the [`Command Palette`](https://www.sumatrapdfreader.org/docs/Command-Palette)?  I can see the following nice orders:

- Last chronologically used (per window, in the order of last chronologically used windows).
- Alphabetical in file names.
- Alphabetical in full file names.
- Left-to-right (per window, in the order of last chronologically used windows).

**Comment by azurtem** [comment]
Hi
Thanks for a great tool
I'm sure it has been asked a million times before but for those who have a tendency to open a trillion pdf files at the same time it would be pratical if we had an option to close all tabs with one click, as they have with browsers, or else close all tabs but this one
regards
yann

**Reply by GitHubRulesOK** [reply]
@azurtem Like this ?
![image](https://user-images.githubusercontent.com/12754913/205135645-ca6ef294-bb0f-44aa-a25f-e4489a414c38.png)

**Comment by tactilis** [comment]
@azurtem
 
![image](https://user-images.githubusercontent.com/4060092/205107953-50117772-d6d2-4bef-be4c-fe3c8d9786c8.png)
 
![image](https://user-images.githubusercontent.com/4060092/205108226-bcd0a9b8-9532-4051-96db-a1a667971fb5.png)

**Reply by GitHubRulesOK** [reply]
sorry did not spot you already answered thanks

**Reply by goyalyashpal** [reply]
@tactilis please use the reply feature for such things, rather than creating new threads

**Reply by kerryland** [reply]
我很好奇你为什么这么说。 您需要什么功能？

**Reply by GitHubRulesOK** [reply]
Haigh ansin!
Perhaps more formally
Dia Daoibh Domhan!

**Comment by goyalyashpal** [comment]
What does DDE mean?

**Reply by tactilis** [reply]
[https://learn.microsoft.com/en-us/search/?terms=DDE](https://learn.microsoft.com/en-us/search/?terms=DDE)

**Reply by goyalyashpal** [reply]
wow nice, thanks you both @ilyaza and @ tactilis

**Comment by ilyaza** [comment]
On Sun, Dec 18, 2022 at 03:42:44AM -0800, yashpalgoyal1304 wrote:
> What does DDE mean?

Eh?!

  https://www.sumatrapdfreader.org/docs/DDE-Commands

Hope this helps,
Ilya

**Comment by wdelacruznoob** [comment]
can you add copy as path when you right click in address bar?

**Comment by GitHubRulesOK** [comment]
@wdelacruznoob
It does that or if want less then show infolder
![image](https://user-images.githubusercontent.com/12754913/218287229-831c269d-ff6b-49a6-8cbd-3f3976b42335.png)

**Reply by wdelacruznoob** [reply]
![image](https://user-images.githubusercontent.com/125164530/218287339-412268f5-900e-45e0-9e90-2ad854df6fed.png)
i know it does. however, in foxit reader, it can copy the path in 2 clicks. Regardless, sumatrapdf is a great an lightweight pdf tool

**Reply by GitHubRulesOK** [reply]
Well if we are counting SumatraPDF could do it on one key while standing on its head and shooting cans off a wall, but then what use is that without a target,

Look at set any custom key to call CmdOpenWithExplorer = current path however have to admit its one short of a full load !
double click with synctex can be set to call with current filename but requires a dummy file for that function so not a workable method either

**Comment by daysneco** [comment]
![image](https://user-images.githubusercontent.com/6323093/221350846-3db4da49-bc22-40f3-b80e-b3860bf12ed2.png)
why i can't download

**Reply by GitHubRulesOK** [reply]
because the dynamic data is scriped in js and you seem to not be using js scripts on that page., so whilst I too detest scripted web pages sadly 99.5% of web pages have some script belong a google.

**Comment by bengit2002** [comment]
I would request you to include the themes/skin into SumatraPDF: the light theme/skin, the dark theme/skin and the system theme. That would give users option to choose dark theme or system theme to suit their need (I think dark mode is better for my eyes). 

Thanks and regards.

**Reply by GitHubRulesOK** [reply]
@bengit2002 
SumatraPDF is minimalist so no theming other than tabs which was working well enough in Windows Vista/XP but then MS introduced Metro theming and it was downhill after that so current theming is  variable to say the least
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a9a1bb99-caa0-4fdc-bdad-69f5f986617c)

**Reply by bengit2002** [reply]
Thanks Git. I would appreciate if I'll see it in the next update. That would make the software more convenient.

This screenshot is taken from Foxit Reader in Skin option.

![FoxitReaderSkins](https://github.com/sumatrapdfreader/sumatrapdf/assets/142773299/14fd21c8-d011-47d3-9749-8cf68cdea6a4)

**Reply by goyalyashpal** [reply]
personally the theme wouldnt have mattered much, as the sumatra's elements occupied small area. 

but now with standalone home page, command palette etc 
yes, themes would be nice

**Comment by ghost** [comment]
will there be no new versions anymore?

**Reply by kerryland** [reply]
Yes there will. kjk is working hard on it, right now, every day.

**Reply by GitHubRulesOK** [reply]
@Lukjan39 
Note that @kerryland 
Has also been a big influence on several recent improvements including aspects of annotation.

**Comment by ghost** [comment]
Will it be possible to fill out forms in PDF?

**Reply by kjk** [reply]
not it this release

**Reply by ghost** [reply]
:(

**Reply by GitHubRulesOK** [reply]
@Lukjan39 
Forms actually comprise many types, so to be clear the Higher level "Adobe only" forms would not be possible in many readers
Even Edge using Acrobat enhancements cannot open some Adobe non ISO standard types (many as used by Gov / Org web sites)
Forms start with 
1) a basic "AcroForm" without Javascript, these should be the standard, to aim for.
2) then Basic Enhanced by Scripting (To be avoided)
3) Static Adobe XFA (With / Without AcroForm) Should be classed as retiring.
4) Dynamic XFA (to be only filled by Adobe subscription products) Commonly favoured by large  organisations as enforces their choices.

**Reply by ghost** [reply]
ok

**Comment by bengit2002** [comment]
SumatraPDF is now one of the best PDF viewers for Windows. Thank the author for listening to users!
